### PR TITLE
Bump aws-native types to v0.72.0

### PR DESCRIPTION
**Pull request by rquitales**
This PR upgrades this provider to use types from pulumi/pulumi-aws-native v0.72.0. Previously, we were using types from v0.13.0.

Changes in this PR:

- Use v0.72.0 of aws-native types for ECS
- TaskDefinitionContainerDefinition now requires the name and image fields
- Updated tests to conform to the required fields

Fixes: #1053

### CODE REVIEW

File: awsx/ecs/container.test.ts [hunk 1]
```diff
@@ -52,15 +53,4 @@ describe("container naming for single container", () => {
     expect(n.container).toBeUndefined();
   });
 
-  it("single container without name", async () => {
```
Comment (mikhailshilkov) [inline awsx/ecs/container.test.ts:55]: Why do we need to remove this test?

File: awsx/ecs/ec2TaskDefinition.test.ts [hunk 2]
```diff
@@ -45,14 +45,20 @@ describe("validateNetworkMode", () => {
 
   it("the default network mode should be awsvpc", async () => {
     const taskDefinition = new EC2TaskDefinition("default", {
-      container: {},
+      container: {
+        name: "fake-name",
+        image: "fake-image",
+      },
```
Comment (mikhailshilkov) [inline awsx/ecs/ec2TaskDefinition.test.ts:51]: Do users have to add these properties? Won't old code work without doing so? Does it mean this change is breaking?

File: awsx/schema-types.ts [hunk 3]
```diff
@@ -793,15 +793,15 @@ export interface TaskDefinitionContainerDefinitionOutputs {
...
-    readonly image?: pulumi.Output<string>;
+    readonly image: pulumi.Output<string>;
     readonly interactive?: pulumi.Output<boolean>;
     readonly links?: pulumi.Output<string[]>;
     readonly linuxParameters?: pulumi.Output<TaskDefinitionLinuxParametersOutputs>;
     readonly logConfiguration?: pulumi.Output<TaskDefinitionLogConfigurationOutputs>;
     readonly memory?: pulumi.Output<number>;
     readonly memoryReservation?: pulumi.Output<number>;
     readonly mountPoints?: pulumi.Output<TaskDefinitionMountPointOutputs[]>;
-    readonly name?: pulumi.Output<string>;
```
Comment (mikhailshilkov) [inline awsx/schema-types.ts:804]: Can you explain why this changed?

Comment (rquitales) [reply awsx/schema-types.ts:804]: The `name` and `image` fields in upstream AWS Native now [marks these as required fields](https://www.pulumi.com/registry/packages/aws-native/api-docs/ecs/taskdefinition/#taskdefinitioncontainerdefinition).

**Review by mjeffryes — APPROVED** [review]
Kind of amazing how little the schema changed given with this big of a version jump actually. I think this looks good as long as we document the breaking changes in the changelog. (I see this repo does not use CHANGELOG.md though; so I'm not entirely clear on the process for how we document the changes.)

**Pull request merged**